src/datasets/tasks/text_classification.py

```diff
@@ -1,30 +1,26 @@
 from dataclasses import dataclass
-from typing import Dict, List
+from typing import ClassVar, Dict, List
 
 from ..features import ClassLabel, Features, Value
 from .base import TaskTemplate
 
 
 @dataclass(frozen=True)
 class TextClassification(TaskTemplate):
-    task = "text-classification"
-    input_schema = Features({"text": Value("string")})
+    task: ClassVar[str] = "text-classification"
+    input_schema: ClassVar[Features] = Features({"text": Value("string")})
     # TODO(lewtun): Since we update this in __post_init__ do we need to set a default? We'll need it for __init__ so
     # investigate if there's a more elegant approach.
-    label_schema = Features({"labels": ClassLabel})
+    label_schema: ClassVar[Features] = Features({"labels": ClassLabel})
     labels: List[str]
     text_column: str = "text"
     label_column: str = "labels"
 
     def __post_init__(self):
-        assert sorted(set(self.labels)) == sorted(self.labels), "Labels must be unique"
+        assert len(self.labels) == len(set(self.labels)), "Labels must be unique"
         # Cast labels to tuple to allow hashing
-        object.__setattr__(self, "labels", tuple(sorted(self.labels)))
-        object.__setattr__(self, "text_column", self.text_column)
-        object.__setattr__(self, "label_column", self.label_column)
+        self.__dict__["labels"] = tuple(sorted(self.labels))
         self.label_schema["labels"] = ClassLabel(names=self.labels)
```

Comment:
This line is an issue. Modifying a class attribute based on the value of an instance attribute doesn't make sense if multiple instances of the same class are allowed (if that's the case, the class attribute will have a valid value only for the instance that was initialized last). One way to fix this is with the help of descriptors:
```python
class FeaturesWithLazyClassLabel:
    def __init__(self, features, label_column="labels"):
        assert label_column in features, f"Key '{label_column}' missing in features {features}"
        self._features = features
        self._label_column = label_column

    def __get__(self, obj, objtype=None):
        if obj is None:
            return self._features
     
        assert hasattr(obj, self._label_column), f"Object has no attribute '{self._label_column}'"
        # note: this part is not cached, but we can add this easily
        features = self._features.copy()
        features["labels"] = ClassLabel(names=getattr(obj, self._label_column))
        return features
```

and then in `TextClassification`:

```python
label_schema: ClassVar[Features] = FeaturesWithLazyClassLabel(Features({"labels": ClassLabel}))
```

Reply:
+1 on this !

The fix using `FeaturesWithLazyClassLabel` is fine IMO

Reply:
nice catch @mariosasko! i'm also happy with the `FeatureWithLazyClassLabel` fix for now

Reply:
Yes, this is a temporary solution. I agree we should try to find a simpler way to fix this. The current API design does not seem to fit the task very well (having `label_schema` defined as a class attribute but depends on the value of instances?), so we should rethink that part IMO.